### PR TITLE
doc(headless): add relative date facet to samples

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-843

### CODE REVIEW

File: packages/samples/headless-react/src/components/date-facet/date-facet.class.tsx [hunk 1]
```diff
@@ -51,7 +51,11 @@ export class DateFacet extends Component<DateFacetProps, DateFacetState> {
       return null;
     }
 
-    if (!this.state.values.length) {
+    if (
```
Comment (ThibodeauJF) [inline packages/samples/headless-react/src/components/date-facet/date-facet.class.tsx:54]: Range facets have values with 0 results, we have to check if there are selected or values with results

**Review by nbuissoncoveo — APPROVED** [review]

File: packages/samples/headless-react/src/components/relative-date-facet/relative-date-facet.fn.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ *         start: {period: 'past', unit: 'month', amount: 1},
+ *         end: {period: 'now'},
+ *       }),
+ *       buildDateRange({
+ *         start: {period: 'past', unit: 'quarter', amount: 1},
+ *         end: {period: 'now'},
+ *       }),
+ *       buildDateRange({
+ *         start: {period: 'past', unit: 'year', amount: 1},
+ *         end: {period: 'now'},
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/relative-date-facet/relative-date-facet.fn.tsx:86]: To keep the reference short, I would suggest showing only two ranges.

File: packages/samples/headless-react/src/components/relative-date-facet/relative-date-facet.fn.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+export const RelativeDateFacet: FunctionComponent<RelativeDateFacetProps> = (
+  props
+) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  function getKeyForRange(value: DateFacetValue) {
+    return `[${value.start}..${value.end}${value.endInclusive ? ']' : '['}`;
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/relative-date-facet/relative-date-facet.fn.tsx:21]: Could we simplify the key to just use `start` and `end`?

File: packages/samples/headless-react/src/components/relative-date-facet/relative-date-facet.fn.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import {useEffect, useState, FunctionComponent} from 'react';
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/relative-date-facet/relative-date-facet.fn.tsx:1]: Let's update the date-facet config under the `doc-parser` directory to reference the new sample files.

**Review by samisayegh — APPROVED** [review]

File: packages/samples/headless-react/src/components/date-facet/date-facet.class.tsx (hunk 1 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/date-facet/date-facet.class.tsx:54]: I think we should extract this into a separate function and possibly add a comment explaining this.

**Review by btaillon — APPROVED** [review]

**Pull request merged**